### DISCUSSION TITLE
"matchGroupNames" for extra settings for given group

### DISCUSSION THREAD

**Discussion by szpak** [renovatebot/renovate#20367] [Suggest an Idea]
### What would you like Renovate to be able to do?

Having some group defined (manually or derived from some preset), I would like to be able to define/override some settings, e.g.

```
{
  "packageRules": [
    {
      "matchGroupNames": ["spock"],    //defined in some preset
      "matchCurrentVersion": "/-groovy-3\\.0$/",
      "allowedVersions": "/-groovy-3\\.0$/"
    }
  ]
}
```

### If you have any ideas on how this should be implemented, please tell us here.

A new `matchGroupNames` field - available within the "packageRules" could be added.

Triggered by the [recent discussion](https://github.com/renovatebot/renovate/discussions/20359#discussioncomment-4949479).

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
Although `matchGroupNames` is in theory possible, the rest of what you ask for is not possible - or at least not a good idea to for us to support. I have converted this back to a discussion because it would ultimately be a bad idea to support this as described.

Branch names are only assigned _after_ updates are decided. Updates are decided _after_ `allowedVersions`. By the time the branch name is assigned, it's too late to apply `allowedVresions`
